### PR TITLE
Add transient error retry

### PR DESCRIPTION
**Pull request by ZakharMakarevich**
Add transient built-in error retry mechanism based on [Polly](https://github.com/App-vNext/Polly) lib

### CODE REVIEW

File: Softeq.NetKit.Services.PushNotifications/Softeq.NetKit.Services.PushNotifications.csproj [hunk 1]
```diff
@@ -16,7 +16,8 @@
 
   <ItemGroup>
     <PackageReference Include="Ensure.That" Version="8.1.1" />
-    <PackageReference Include="Microsoft.Azure.NotificationHubs" Version="3.3.0" />
+    <PackageReference Include="Microsoft.Azure.NotificationHubs" Version="4.1.0" />
+    <PackageReference Include="Polly" Version="7.2.1" />
```
Comment (smolyakoff) [inline Softeq.NetKit.Services.PushNotifications/Softeq.NetKit.Services.PushNotifications.csproj]: If we upgrade the `Microsoft.Azure.NotificationHubs`, can't we use the built-in resiliency measures? I'm looking at the source code and see that it does have a built-in retry policy: https://github.com/Azure/azure-notificationhubs-dotnet/blob/main/src/Microsoft.Azure.NotificationHubs/NotificationHubClient.cs#L2836

So maybe we need to expose a knob from the wrapper library if it's not enabled by default.

**Review by smolyakoff — CHANGES_REQUESTED** [review]
A better option would be to use built-in resiliency from the Notification Hub SDK if it's possible, without introducing another dependency on Polly.

File: Softeq.NetKit.Services.PushNotifications/Softeq.NetKit.Services.PushNotifications.csproj (hunk 1 above)
Comment (ZakharMakarevich) [reply Softeq.NetKit.Services.PushNotifications/Softeq.NetKit.Services.PushNotifications.csproj]: Built-in retry mechanism utilized instead Polly

File: Softeq.NetKit.Services.PushNotifications/Client/AzureNotificationHubSender.cs [hunk 2]
```diff
@@ -21,7 +21,10 @@ public class AzureNotificationHubSender : IPushNotificationSender
         public AzureNotificationHubSender(AzureNotificationHubConfiguration configuration)
         {
             Ensure.That(configuration, nameof(configuration)).IsNotNull();
-            _hub = NotificationHubClient.CreateClientFromConnectionString(configuration.ConnectionString, configuration.HubName);
+            _hub = new NotificationHubClient(configuration.ConnectionString, configuration.HubName, new NotificationHubSettings
+            {
+                RetryOptions = configuration.NotificationHubRetryOptions
```
Comment (smolyakoff) [inline Softeq.NetKit.Services.PushNotifications/Client/AzureNotificationHubSender.cs]: Let's expose the whole `NotificationHubSettings` object on `AzureNotificationHubConfiguration`? Don't see the reason why we should limit the ability to configure the underlying client to just retries.

Comment (ZakharMakarevich) [reply Softeq.NetKit.Services.PushNotifications/Client/AzureNotificationHubSender.cs]: The whole configuration is exposed now :)

**Review by smolyakoff — APPROVED** [review]
LGTM

**Review by eddyable — APPROVED** [review]

**Pull request merged**